src/uprobe/mod.rs

```diff
@@ -79,17 +88,13 @@ impl Uprobe {
                 message: "binary path is invalid".to_string(),
             });
         }
-        if self.symbol.is_none() {
-            return Err(BccError::InvalidUprobe {
-                message: "symbol is required".to_string(),
-            });
-        }
         let binary = binary.unwrap();
-        let symbol = self.symbol.unwrap();
+        let symbol = self.symbol.as_ref().map(|s| s.as_str()).unwrap_or("");
```

Comment:
This feels kinda weird?

Reply:
Okay, good not just me imagining it:
```
warning: called `.as_ref().map(|s| s.as_str())` on an Option value. This can be done more directly by calling `self.symbol.as_deref()` instead
  --> src/uprobe/mod.rs:92:22
   |
92 |         let symbol = self.symbol.as_ref().map(|s| s.as_str()).unwrap_or("");
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try using as_deref instead: `self.symbol.as_deref()`
   |
   = note: `#[warn(clippy::option_as_ref_deref)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#option_as_ref_deref
```